README.md

```diff
@@ -1,94 +1,60 @@
 
-### Result pipelines `~>>`
+### `OK.with`
 
-This macro allows pipelining result tuples through a pipeline of functions.
-The `~>>` macro is the is equivalent to bind/flat_map in other languages.
+`OK.with` allows for more concise and ultimately more readable code than the native `with` construct. It does this by leveraging result monads for both the happy and non-happy paths. By extracting the actual function return values from the result tuples, `OK.with` reduces noise which improves readability and recovers precious horizontal code real estate. This also encourages writing idiomatic Elixir functions which return `:ok`/`:error` tuples.
 
-```elixir
-import OK, only: ["~>>": 2]
+* [Elegant error handling with result monads, alternative to elixir `with` special form](https://elixirforum.com/t/elegant-error-handling-with-result-monads-alternative-to-elixir-with-special-form/3264/1)
+* [Discussion on :ok/:error](https://elixirforum.com/t/usage-of-ok-result-error-vs-some-result-none/3253)
 
-def get_employee_data(file, name) do
-  {:ok, file}
-  ~>> File.read
-  ~>> Poison.decode
-  ~>> Dict.fetch(name)
-end
-
-def handle_user_data({:ok, data}), do: IO.puts("Contact at #{data["email"]}")
-def handle_user_data({:error, :enoent}), do: IO.puts("File not found")
-def handle_user_data({:error, {:invalid, _}}), do: IO.puts("Invalid JSON")
-def handle_user_data({:error, :key_not_found}), do: IO.puts("Could not find employee")
-
-get_employee_data("my_company/employees.json")
-|> handle_user_data
-```
-
-Code structured like this is an example of [railway programming](http://www.zohaib.me/railway-programming-pattern-in-elixir/).
-
-[Forum discussion on error handling in pipelines](https://elixirforum.com/t/ok-elegant-error-handling-for-elixir-pipelines-version-1-0-released/1932)
-
-### Result blocks `with`
+#### Basic Usage
 
-For situations when the pipeline macro is not sufficiently flexible.
+* Use the `<-` operator to match & extract a value for an `:ok` tuple.
+* Use the `=` operator as you normally would for pattern matching an untagged result.
+* The last line should either be a function that returns the tuple, or the literal tuple itself.
 
-To extract a value for an ok tuple use the `<-` operator.
+_NB: Statements are **not** delimited by commas as with the native Elixir `with` construct._
 
 ```elixir
 require OK
 
 OK.with do
-  user <- fetch_user(1)
-  cart <- fetch_cart(1)
-  order = checkout(cart, user)
-  save_order(order)
+  user <- fetch_user(1)        # `<-` operator means func returns {:ok, user}
+  cart <- fetch_cart(1)        # `<-` again, {:ok, cart}
+  order = checkout(cart, user) # `=` allows pattern matching on non-tagged funcs
+  save_order(order)            # Returns a tuple.
 end
 ```
 
-`Ok.with/1` supports an else block that can be used for handling error values.
+The above could also be written as
```

Comment:
It might be best to leave the optional part out, but I personally have gotten very fond of the `do` return section in the native `with` block that makes it explicit on what exactly is being returned. I definitely need to reword these though, and I'll just go ahead and take this out.